bin/validate.py

```diff
@@ -329,6 +337,46 @@ def make_products(dataset, target_dir, gaf_path, products, ontology_graph):
 
     return [product_files[prod].name for prod in sorted(product_files.keys()) if products[prod]]
 
+
+def make_gpads(dataset, gaf_path, products, ontology_graph, noctua_gpad_file):
+    if not products["gpad"]:
+        # Bail if we have no products
+        return []
+
+    with open(gaf_path) as sg:
+        lines = sum(1 for line in sg)
+
+    # logger.info("AssocParserConfig used: {}".format(config))
+    split_source = os.path.split(gaf_path)[0]
+    validated_gpad_path = os.path.join(split_source, "{}_gpad.gpad".format(dataset))
+    outfile = open(validated_gpad_path, "w")
+    gpadwriter = GpadWriter(file=outfile, version="GPAD_2_0")
+
+    print("noctua-gpad-file: {}".format(noctua_gpad_file))
+    if noctua_gpad_file is not None:
+        with open(noctua_gpad_file) as nf:
+            gpadparser = GpadParser(config=assocparser.AssocParserConfig(
+                ontology=ontology_graph,
+                paint=False,
+            ))
+
+            click.echo("Making noctua gpad products...")
+            with click.progressbar(iterable=gpadparser.association_generator(file=nf), length=lines) as associations:
```

Comment:
The only nit-picky thing that you wouldn't really need to change (or I'm reading it wrong) is this `lines` might be the count of lines in `gaf_path` above (line 347) and not `noctua_gpad_file` as I think is intended.